staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -691,6 +691,33 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 	}
 }
 
+// requestPreflightCheck looks for common programmer errors on Request.
+//
+// We tackle here two programmer mistakes. The first one is to try to create
+// something(POST) using an empty string as namespace with namespaceSet as
+// true. If namespaceSet is true then namespace should also be defined. The
+// second mistake is, when under the same circumstances, the programmer tries
+// to GET, PUT or DELETE a named resource(resourceName != ""), again, if
+// namespaceSet is true then namespace must not be empty.
+func (r *Request) requestPreflightCheck() error {
+	if !r.namespaceSet {
+		return nil
+	}
+	if len(r.namespace) > 0 {
+		return nil
+	}
+
+	switch r.verb {
+	case "POST":
+		return fmt.Errorf("an empty namespace may not be set during creation")
```

Comment:
If you want a typed error, why not just return the typed error from here?

Reply:
The documentation/comment stated that programmer would receive a typed error in case of errors **prior** to a call to `Do()`. As far as I could see this is not true hence my attempt to make the code comply with the documentation/comment. 

I have sent a commit that removes the misguiding comment and restored almost everything to its place. PTAL.